src/AEG/AEGUtils.ts

```diff
@@ -154,22 +172,23 @@ function edgesWithin(rect1: Rectangle, rect2: Rectangle): boolean {
 export function pointInRect(rect: Rectangle, point: Point): boolean {
     const rectCorners = rect.getCorners();
 
-    //Points on edges are considered to be contained
+    //Points on edges are considered to be intersected, not contained
     return (
-        rectCorners[0].x <= point.x &&
-        rectCorners[1].x >= point.x &&
-        rectCorners[0].y <= point.y &&
-        rectCorners[2].y >= point.y
+        rectCorners[0].x < point.x &&
+        rectCorners[1].x > point.x &&
+        rectCorners[0].y < point.y &&
+        rectCorners[2].y > point.y
     );
 }
 
 /**
- * Method that checks whether a point is within the given ellipse.
+ * Method that returns the value of a point compared to the ellipse
```

Comment:
"Value compared to a point" is not a clear explanation of what this value represents, my suggestion is to rework this to be "signedDistanceFromEllipse" by moving the one to the other side of the equation:
 ```
 (x-h)^2/rx^2 + (y-k)^2/ry^2 <= 1
 (x-h)^2/rx^2 + (y-k)^2/ry^2 - 1 <= 0
  ```
 Now this is a signed distance metric, equal to 0 means the point is on the ellipse (no displacement). <0 means inside the ellipse and >0 means outside the ellipse. 

Note that this is not a distance metric since distance is always positive, signed distance is directed and is actually based off of the topological notion of a point being inside of something! Perfect for our application.

https://en.wikipedia.org/wiki/Signed_distance_function

Reply:
Albeit a true signed distance function is positive if the point is inside and negative outside